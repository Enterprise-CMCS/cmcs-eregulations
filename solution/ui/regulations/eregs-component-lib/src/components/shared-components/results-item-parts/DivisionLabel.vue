<script setup>
defineProps({
    division: {
        type: Object,
        required: true,
    },
});
</script>

<template>
    <span
        class="result__context--division result__context--division--bar"
        :title="`${division?.group?.name} — ${division?.name}`"
        data-testid="division-label"
    >Uploaded by {{ division?.group?.abbreviation }} —
        {{ division?.abbreviation }}</span>
</template>
