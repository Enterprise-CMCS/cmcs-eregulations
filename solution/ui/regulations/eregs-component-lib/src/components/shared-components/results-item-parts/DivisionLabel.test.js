import { render } from "@testing-library/vue";
import { describe, it, expect } from "vitest";

import policyDocsFixture from "cypress/fixtures/policy-docs.json";

import DivisionLabel from "./DivisionLabel.vue";

describe("DivisionLabel", () => {
    it("Renders a Division Label component", async () => {
        const wrapper = render(DivisionLabel, {
            props: {
                division: policyDocsFixture.results[1].division,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
