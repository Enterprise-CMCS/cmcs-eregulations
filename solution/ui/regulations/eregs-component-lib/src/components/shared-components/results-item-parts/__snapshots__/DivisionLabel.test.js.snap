// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DivisionLabel > Renders a Division Label component 1`] = `
{
  "baseElement": <body>
    <div>
      <span
        class="result__context--division result__context--division--bar"
        title="Mock Group 1 — Mock Division 1"
      >
        Uploaded by MG1 — MD1
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="result__context--division result__context--division--bar"
      title="Mock Group 1 — Mock Division 1"
    >
      Uploaded by MG1 — MD1
    </span>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;
